loop_stack.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+set -eux
+
+# root directory for the repository
+export HPC_STACK_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 -p <prefix> | -c <config> | -y <yaml> -h"
+  echo
+  echo "  -p  installation prefix <prefix>    DEFAULT: $HOME/opt"
+  echo "  -c  use configuration file <config> DEFAULT: config/config_custom.sh"
+  echo "  -y  use yaml file <yaml>            DEFAULT: config/stack_custom.yaml"
+  echo "  -h  display this message and quit"
+  echo
+  exit 1
+}
+
+# ==============================================================================
+
+[[ $# -eq 0 ]] && usage
```

Comment:
That is not sufficient.  You also need a yaml file, since you don't define a default.  It is in the usage, but not in the script.

Reply:
Actually, you don't need this check at ll, since all the default options are pre-defined.